tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -121,17 +134,55 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
 		ginkgo.By("create 3 SPBM policies with thin, LZT, EZT volume allocation respectively")
 		ginkgo.By("create 3 storage classes, each with a SPBM policy created from step 1")
 		ginkgo.By("create a PVC each using the storage policies created from step 2")
+		var storageclass *storagev1.StorageClass
+		var pvclaim *v1.PersistentVolumeClaim
+		var err error
+		var policyName string
+		var policyID *pbmtypes.PbmProfileId
 		for _, at := range allocationTypes {
-			policyID, policyName := createVmfsStoragePolicy(
+			policyID, policyName = createVmfsStoragePolicy(
 				ctx, pc, at, map[string]string{categoryName: tagName})
+
 			defer func() {
 				deleteStoragePolicy(ctx, pc, policyID)
 			}()
-			scParameters[scParamStoragePolicyName] = policyName
 			policyNames = append(policyNames, policyName)
-			storageclass, pvclaim, err := createPVCAndStorageClass(client,
-				namespace, nil, scParameters, "", nil, "", false, "")
-			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		if supervisorCluster {
+			assignPolicyToWcpNamespace(client, ctx, namespace, policyNames)
+		} else if guestCluster {
+			_, svNamespace := getSvcClientAndNamespace()
+			assignPolicyToWcpNamespace(client, ctx, svNamespace, policyNames)
+		}
+
+		for _, policyName := range policyNames {
+			if vanillaCluster {
+				ginkgo.By("CNS_TEST: Running for vanilla k8s setup")
+				scParameters[scParamStoragePolicyName] = policyName
+				storageclass, pvclaim, err = createPVCAndStorageClass(client,
+					namespace, nil, nil, "", nil, "", false, "")
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			} else if supervisorCluster {
+				ginkgo.By("CNS_TEST: Running for WCP setup")
+				profileID := e2eVSphere.GetSpbmPolicyID(policyName)
+				scParameters[scParamStoragePolicyID] = profileID
+				// create resource quota
+				createResourceQuota(client, namespace, rqLimit, policyName)
+				storageclass, pvclaim, err = createPVCAndStorageClass(client, namespace, nil,
+					scParameters, "", nil, "", false, "", policyName)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			} else {
+				ginkgo.By("CNS_TEST: Running for GC setup")
+				createResourceQuota(client, namespace, rqLimit, policyName)
+				scParameters[svStorageClassName] = policyName
+				storageclass, err = client.StorageV1().StorageClasses().Get(ctx, policyName, metav1.GetOptions{})
+				if !apierrors.IsNotFound(err) {
```

Comment:
if SC is not available we should fail here itself right?

Reply:
Done